scylla-gdb.py

```diff
@@ -5064,9 +5061,35 @@ def invoke(self, ptr):
         if m is None:
             raise ValueError("Failed to extract type name from symbol name `{}'".format(symbol_name))
 
-        actual_type = gdb.lookup_type(m.group(1)).pointer()
-        return ptr.reinterpret_cast(actual_type)
+        return m
+
+    def invoke(self, ptr):
+        if not isinstance(ptr, gdb.Value):
+            ptr = gdb.parse_and_eval(ptr)
+
+        m = self._check_vptr(ptr)
+
+        actual_type = gdb.lookup_type(m.group(1))
+        actual_type_ptr = actual_type.pointer()
+
+        if int(m.group(2)) == 16:
+            return ptr.reinterpret_cast(actual_type_ptr)
```

Comment:
Is this 16 correct for ARM and other archs too? Is there a less internals dependent way to code this?

Reply:
Scratch that. The 16 is already present in the code, this patch doesn't introduce it. So this thing can't be a regression.

Reply:
To be honest I have no idea. Most (if not all) of the content of this file is based on observation and intuition built up by working with coredumps.

Reply:
As it should, it really should be updated on demand and generalized when the need arises, not before.